### ISSUE TITLE
Allow opt-in plaintext uploaderId for journal schema

### ISSUE THREAD

**Issue by corecache** [EDSM-NET/EDDN#77]
The uploaderId is currently beeing scrambled every 12 hours and there is no option to avoid that.

The uploaderId should be scrambed and regenerated by default, but an additional attribute/option in the uploaded document should allow to disable this feature for those who want their Plaintext uploaderId visible to the public.

**Comment by AnthorNet** [comment]
Hi, for what purpose?

**Comment by corecache** [comment]
We used to have a site where our guild members told us their uploaderID - which we then tracked to build up statistics for our guild leaderboard, where they could earn monthly decals and prizes. This feature sadly broke when scrambling uploaderID.

**Comment by corecache** [comment]
Something along the lines of:
` if ('allowPublicID' in json['header']) and (json['header']['allowPublicID'] == True):`

**Comment by inorton** [comment]
I have been thinking about this, @corecache that might work.

I was also thinking about a way of doing this with some simple crypto, but maybe that's overkill

**Comment by inorton** [comment]
Hmm.  Ok.. so bit of a problem here then. 

- Some people don't mind being identified
- Some people actually need to be able to identify the data they submitted
- By default, we should continue to mask identities

**Comment by corecache** [comment]
@inorton That is a nice solution there, good work

**Comment by corecache** [comment]
@AnthorNet So, everything is ready and in place

**Comment by AnthorNet** [comment]
I'm still not sure, as already stated, EDDN is not meant to transfer personal information.

**Comment by inorton** [comment]
Yeah, I'm in two minds about it.  Perhaps there is scope for some other sort of similar service where people can submit personal records to verify activity? It's a common use case for BGS work

**Comment by inorton** [comment]
Perhaps there is an Inara API that can do that?
